### PR TITLE
C2 mohamed notifications update

### PR DESCRIPTION
**Pull request by mohamedahmedawad93**
@HebaElSobky please review all scenarioes

### CODE REVIEW

**Comment by mohamedahmedawad93** [comment]
@BeshoSmile please review my code

**Comment by mohamedahmedawad93** [comment]
@HebaElSobky @BeshoSmile 
dummy code:

from tager_www.models import *
user1=UserProfile(name='Mohamed',email='1@3.com',password='123');user1.save()
user2=UserProfile(name='Amr',email='2@3.com',password='123');user2.save()
channel1=Channel(name='cars',description='jkj');channel1.save()
subchannel1=SubChannel(name='sport cars',channel=channel1);subchannel1.save()
subchannel2=SubChannel(name='SUV',channel=channel1);subchannel2.save()
attribute1=Attribute(name='color',subchannel=subchannel1,weight=1);attribute1.save()
attribute2=Attribute(name='color',subchannel=subchannel2,weight=1);attribute2.save()
choice1=AttributeChoice(attribute_id=attribute1, value='red');choice1.save()
choice2=AttributeChoice(attribute_id=attribute2, value='red');choice2.save()
subscription1=Subscription(channel=channel1,sub_channel=subchannel1,parameter=attribute1,choice=choice1);subscription1.save()
subscription2=Subscription(channel=channel1,sub_channel=subchannel1);subscription2.save()
subscription3=Subscription(channel=channel1);subscription3.save()
subscription4=Subscription(channel=channel1,sub_channel=subchannel2,parameter=attribute2,choice=choice2);subscription4.save()
subscription5=Subscription(channel=channel1,sub_channel=subchannel2);subscription5.save()
channel2=Channel(name='Anaimals',description='jkj');channel2.save()
subchannel3=SubChannel(name='Cats',channel=channel2);subchannel3.save()
subchannel4=SubChannel(name='Dogs',channel=channel2);subchannel4.save()
attribute3=Attribute(name='age',subchannel=subchannel3,weight=1);attribute3.save()
attribute4=Attribute(name='age',subchannel=subchannel4,weight=1);attribute4.save()
choice3=AttributeChoice(attribute_id=attribute3, value='2');choice3.save()
choice4=AttributeChoice(attribute_id=attribute4, value='2');choice4.save()
subscription6=Subscription(channel=channel2,sub_channel=subchannel3,parameter=attribute3,choice=choice3);subscription6.save()
subscription7=Subscription(channel=channel2,sub_channel=subchannel3);subscription7.save()
subscription8=Subscription(channel=channel2);subscription8.save()
subscription9=Subscription(channel=channel2,sub_channel=subchannel4,parameter=attribute4,choice=choice4);subscription9.save()
subscription10=Subscription(channel=channel2,sub_channel=subchannel4);subscription10.save()
P1= Post(state="New",expired=False,title="Title of post 1",is_hidden=False,quality_index=67,price=100,is_sold=False,subchannel=subchannel1,seller=user1,no_of_reports=0,pub_date=datetime.now());P1.save();
P2= Post(state="New",expired=False,title="Title of post 2",is_hidden=False,quality_index=80,price=100,is_sold=False,subchannel=subchannel1,seller=user1,no_of_reports=0,pub_date=datetime.now());P2.save();
P3= Post(state="New",expired=False,title="Title of post 3",is_hidden=False,quality_index=60,price=100,is_sold=False,subchannel=subchannel2,seller=user1,no_of_reports=0,pub_date=datetime.now());P3.save();
P4= Post(state="New",expired=False,title="Title of post 4",is_hidden=False,quality_index=90,price=100,is_sold=False,subchannel=subchannel2,seller=user1,no_of_reports=0,pub_date=datetime.now());P4.save();
P5= Post(state="New",expired=False,title="Title of post 5",is_hidden=False,quality_index=100,price=100,is_sold=True,subchannel=subchannel3,seller=user1,no_of_reports=0,pub_date=datetime.now());P5.save();
P6= Post(state="New",expired=False,title="Title of post 6",is_hidden=True,quality_index=100,price=100,is_sold=False,subchannel=subchannel3,seller=user2,no_of_reports=0,pub_date=datetime.now());P6.save();
P7= Post(state="New",expired=True,title="Title of post 7",is_hidden=False,quality_index=100,price=100,is_sold=False,subchannel=subchannel4,seller=user2,no_of_reports=0,pub_date=datetime.now());P7.save();
P8= Post(state="New",expired=False,title="Title of post 7",is_hidden=False,quality_index=100,price=130000,is_sold=False,subchannel=subchannel4,seller=user2,no_of_reports=0,pub_date=datetime.now());P8.save();
P9= Post(state="Archived",expired=False,title="Title of post 7",is_hidden=False,quality_index=100,price=130,is_sold=False,subchannel=subchannel1,seller=user2,no_of_reports=50,pub_date=datetime.now());P9.save();

File: project/tager/tager_www/models.py [hunk 1]
```diff
@@ -258,6 +258,8 @@ def calculate_rating(self,rate,post,buyer): #self is the post_owner
 class Notification(models.Model):
     user = models.ForeignKey(UserProfile)
     content = models.CharField(max_length=100)
+    read = models.BooleanField(default=False)
+    url = models.CharField(max_length=50)
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py]: why max length ?

File: project/tager/tager_www/models.py [hunk 2]
```diff
@@ -323,18 +325,23 @@ class Post(models.Model):
     #then we find all users subscribed to all attributes and values of the post and we add it to all_users_subscribed to attributes
     #then we record all notifications in Notification table
     def post_Notification(self):
+        print self
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py:468]: delete unneeded prints

File: project/tager/tager_www/models.py [hunk 3]
```diff
@@ -323,18 +325,23 @@ class Post(models.Model):
...
         all_values_array=[]
         values_array=[]
-        all_values = Value.objects.filter(Post_id = self)
+        all_values = Value.objects.filter(post = self.id)
+        print "look down 1"
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py]: delete unneeded prints

File: project/tager/tager_www/models.py [hunk 4]
```diff
@@ -323,18 +325,23 @@ class Post(models.Model):
...
             values_array.append(value.value)
         attributes_array = []
         for value in all_values_array:
-            attribute  = value.attribute_id
+            attribute  = value.attribute
             attributes_array.append(attribute.name)
-        subchannel_of_post = self.sub_channel_id
-        channel_of_post = subchannel_of_post.channel_id
+        subchannel_of_post = self.subchannel
+        print "look down"
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py]: delete unneeded prints

File: project/tager/tager_www/models.py (hunk 1 above)
Comment (mohamedahmedawad93) [inline project/tager/tager_www/models.py]: char field model should have a maximum length

Comment (mohamedahmedawad93) [inline project/tager/tager_www/models.py]: and content and urls are fixed and it won't exceed those limits

On Sun, Apr 28, 2013 at 11:21 PM, Mohamed Awad <mohamedahmedawad93@gmail.com

> wrote:
> 
> char field model should have a maximum length

File: project/tager/tager_www/views.py [hunk 5]
```diff
@@ -988,3 +990,28 @@ def intrested(request):
...
     return HttpResponse()
 
+#c2-mohamed awad
+#this def returns unread notifications to base.html
+def unread_notifications(request):
+    user_in = request.user
+    all_unread_notifications = Notification.objects.filter(user = user_in, read = False)
+    print all_unread_notifications
+    if all_unread_notifications:
+        print "after if"
```
Comment (BeshoyAtef) [inline project/tager/tager_www/views.py:1474]: delete unneeded prints

File: project/tager/tager_www/views.py [hunk 6]
```diff
@@ -988,3 +990,28 @@ def intrested(request):
 
     return HttpResponse()
 
+#c2-mohamed awad
+#this def returns unread notifications to base.html
+def unread_notifications(request):
+    user_in = request.user
+    all_unread_notifications = Notification.objects.filter(user = user_in, read = False)
+    print all_unread_notifications
```
Comment (BeshoyAtef) [inline project/tager/tager_www/views.py]: delete unneeded prints

File: project/tager/tager_www/views.py [hunk 7]
```diff
@@ -988,3 +990,28 @@ def intrested(request):
...
+    if all_unread_notifications:
+        print "after if"
+        for notification in all_unread_notifications:
+            print "after for"
+            notification.read = True
+            notification.save()
+            print "saved notification"
```
Comment (BeshoyAtef) [inline project/tager/tager_www/views.py]: delete unneeded prints

Comment (HebaElSobky) [inline project/tager/tager_www/views.py]: awad u told me before not to review lsa !!
do u want to be reviewed 5alas ??

**Comment by HebaElSobky** [comment]
@mohamedahmedawad93 do u want to be reviewed or still not yet ??

**Comment by mohamedahmedawad93** [comment]
l2 not yet, ana sayeb de l a5er alsprint

File: project/tager/tager_www/views.py (hunk 7 above)
Comment (mohamedahmedawad93) [inline project/tager/tager_www/views.py]: @HebaElSobky l2 ana sayeb da l a5er alsprint

**Comment by mohamedahmedawad93** [comment]
@BeshoSmile plese review my code

**Comment by mohamedahmedawad93** [comment]
@HebaElSobky please review

**Comment by HebaElSobky** [comment]
Awad ana Msh nasya ana fl gam3a ITPM presentation fa I'll review as soon as
I finish :)

**Comment by mohamedahmedawad93** [comment]
@HebaElSobky 5alas good luck :D

**Comment by mohamedahmedawad93** [comment]
@BeshoSmile please review

**Comment by mohamedahmedawad93** [comment]
@HebaElSobky please review again

**Comment by mohamedahmedawad93** [comment]
@BeshoSmile please review again

**Comment by HebaElSobky** [comment]
awadd GREAT job (Y)

On Thu, May 2, 2013 at 9:05 PM, mohamedahmedawad93 <notifications@github.com

**Comment by HebaElSobky** [comment]
enta 3amaltahaa eno ygeelo notification mn kol l nas, sa7?

**Comment by HebaElSobky** [comment]
@mohamedahmedawad93  scenario verified :+1:

**Comment by mohamedahmedawad93** [comment]
@HebaElSobky now i can sleep peacfully :D

**Comment by HebaElSobky** [comment]
Great job walahiee

**Comment by abdelrahmanMaged** [comment]
t a y 7bebe :O
u forgot that you are required to finish this stroy ::
As a user i should be able to edit my subscriptions :P

**Comment by HebaElSobky** [comment]
t a y ??? daa eh bezabt ??

**Comment by abdelrahmanMaged** [comment]
awad has work to do maenf3sh ynam :D
P.s: notice an d github mafrood our chat maeob2ash hna :D

**Comment by mohamedahmedawad93** [comment]
@AbdelrahmanAyad tharwat 5ad al user story de w mesh rady yseebha malnash feeh b2a :P

**Pull request merged**